tests/transformations/if_extraction_test.py

```diff
@@ -0,0 +1,225 @@
+    subg.add_edge(ifh, if1, InterstateEdge(condition='(flag)', assignments={'outval': 1}))
+    subg.add_edge(ifh, if2, InterstateEdge(condition='(not flag)', assignments={'outval': 2}))
+    subg.add_edge(if1, if3, InterstateEdge())
+    subg.add_edge(if3, if5, InterstateEdge())
+    subg.add_edge(if2, if4, InterstateEdge())
+    subg.add_edge(if4, if5, InterstateEdge())
+    subg.add_edge(if5, if6, InterstateEdge())
+    subg.add_edge(if5, if7, InterstateEdge())
+    subg.add_edge(if6, ift, InterstateEdge())
+    subg.add_edge(if7, ift, InterstateEdge())
+    t0 = ift.add_tasklet('copy', inputs={}, outputs={'__out'}, code='__out = outval')
+    tmp = ift.add_access('tmp')
+    ift.add_memlet_path(t0, tmp, src_conn='__out', memlet=Memlet(expr='tmp[0]'))
+    subg.fill_scope_connectors()
+
+    # Then prepare the parent graph.
+    g = SDFG('prog')
+    g.add_array('A', (10,), dace.float32)
+    g.add_symbol('flag', dace.bool)
+    st0 = g.add_state('outer', is_start_block=True)
+    en, ex = st0.add_map('map', {'i': '0:10'})
+    body = st0.add_nested_sdfg(subg, None, {}, {'tmp'}, symbol_mapping={'flag': 'flag'})
+    A = st0.add_access('A')
+    st0.add_memlet_path(en, body, memlet=Memlet())
+    st0.add_memlet_path(body, ex, src_conn='tmp', dst_conn='IN_A', memlet=Memlet(expr='A[i]'))
+    st0.add_memlet_path(ex, A, src_conn='OUT_A', memlet=Memlet(expr='A[0:10]'))
+    g.fill_scope_connectors()
+
+    return g
+
+
+def make_branched_sdfg_that_depends_on_loop_var():
+    """
+    Construct a simple SDFG that depends on symbols defined or updated in the outer state, e.g., loop variables.
+    """
+    # First prepare the map-body.
+    subg = SDFG('body')
+    subg.add_array('tmp', (1,), dace.float32)
+    subg.add_symbol('outval', dace.float32)
+    ifh = subg.add_state('if_head')
+    if1 = subg.add_state('if_b1')
+    if2 = subg.add_state('if_b2')
+    ift = subg.add_state('if_tail')
+    subg.add_edge(ifh, if1, InterstateEdge(condition='(i == 0)', assignments={'outval': 1}))
+    subg.add_edge(ifh, if2, InterstateEdge(condition='(not (i == 0))', assignments={'outval': 2}))
+    subg.add_edge(if1, ift, InterstateEdge())
+    subg.add_edge(if2, ift, InterstateEdge())
+    t0 = ift.add_tasklet('copy', inputs={}, outputs={'__out'}, code='__out = outval')
+    tmp = ift.add_access('tmp')
+    ift.add_memlet_path(t0, tmp, src_conn='__out', memlet=Memlet(expr='tmp[0]'))
+    subg.fill_scope_connectors()
+
+    # Then prepare the parent graph.
+    g = SDFG('prog')
+    g.add_array('A', (10,), dace.float32)
+    st0 = g.add_state('outer', is_start_block=True)
+    en, ex = st0.add_map('map', {'i': '0:10'})
+    body = st0.add_nested_sdfg(subg, None, {}, {'tmp'})
+    A = st0.add_access('A')
+    st0.add_memlet_path(en, body, memlet=Memlet())
+    st0.add_memlet_path(body, ex, src_conn='tmp', dst_conn='IN_A', memlet=Memlet(expr='A[i]'))
+    st0.add_memlet_path(ex, A, src_conn='OUT_A', memlet=Memlet(expr='A[0:10]'))
```

Comment:
This is really not the right use of `add_memlet_path`. The entire point of adding a memlet _path_ is that you can do it in one go:
`st0.add_memlet_path(body, ex, A, memlet=Memlet('A[i]')`

Reply:
@pratyai

Reply:
I don't think
```python
st0.add_memlet_path(body, ex, A, memlet=Memlet('A[i]'))
```
is equivalent to
```python
st0.add_memlet_path(body, ex, src_conn='tmp', dst_conn='IN_A', memlet=Memlet(expr='A[i]'))
st0.add_memlet_path(ex, A, src_conn='OUT_A', memlet=Memlet(expr='A[0:10]'))
```
I have replaced all the `add_memlet_path()` path calls regardless.